pytest/tests/mocknet/helpers/neard_runner.py

```diff
@@ -665,9 +669,34 @@ def do_ls_backups(self):
         with self.lock:
             return self.data.get('backups', {})
 
-    def do_update_binaries(self):
+    # Updates the URL for the given epoch height or adds a new one if the epoch height does not exit
+    def update_binaries_url(self, neard_binary_url, update_epoch_height):
+        if 'binaries' not in self.config:
+            self.config['binaries'] = []
+
+        if not isinstance(self.config['binaries'], list):
+            self.config['binaries'] = []
+
+        binary = next((b for b in self.config['binaries']
+                       if b['epoch_height'] == update_epoch_height), None)
+        if binary:
+            binary['url'] = neard_binary_url
+        else:
+            self.config['binaries'].append({
+                'url': neard_binary_url,
+                'epoch_height': update_epoch_height
+            })
+
+    def do_update_binaries(self, neard_binary_url, update_epoch_height):
         with self.lock:
             logging.info('update binaries')
+
+            if neard_binary_url and update_epoch_height:
```

Comment:
this is not correct if `update_epoch_height=0`. We do want to update the binary url in that case:

```diff
diff --git a/pytest/tests/mocknet/helpers/neard_runner.py b/pytest/tests/mocknet/helpers/neard_runner.py
index 633b15116..5498382f7 100644
--- a/pytest/tests/mocknet/helpers/neard_runner.py
+++ b/pytest/tests/mocknet/helpers/neard_runner.py
@@ -691,7 +691,7 @@ class NeardRunner:
         with self.lock:
             logging.info('update binaries')
 
-            if neard_binary_url and update_epoch_height:
+            if neard_binary_url is not None and update_epoch_height is not None:
                 logging.info(
                     f'adding new binary url: {neard_binary_url} for epoch {update_epoch_height}'
                 )

```